components/dashboard/src/repositories/detail/prebuilds/PrebuildSettingsForm.tsx

```diff
@@ -0,0 +1,130 @@
+import { useToast } from "../../../components/toasts/Toasts";
+import { SelectInputField } from "../../../components/forms/SelectInputField";
+import { TextInputField } from "../../../components/forms/TextInputField";
+import { WorkspaceClassOptions } from "../shared/WorkspaceClassOptions";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+
+type Props = {
+    configuration: Configuration;
+};
+
+export const PrebuildSettingsForm: FC<Props> = ({ configuration }) => {
+    const { toast } = useToast();
+    const updateConfiguration = useConfigurationMutation();
+
+    const [interval, setInterval] = useState<string>(`${configuration.prebuildSettings?.prebuildInterval ?? 20}`);
+    const [branchStrategy, setBranchStrategy] = useState<BranchMatchingStrategy>(
+        configuration.prebuildSettings?.branchStrategy ?? BranchMatchingStrategy.DEFAULT_BRANCH,
+    );
+    const [branchMatchingPattern, setBranchMatchingPattern] = useState<string>(
+        configuration.prebuildSettings?.branchMatchingPattern || "**",
+    );
+    const [workspaceClass, setWorkspaceClass] = useState<string>(
+        configuration.prebuildSettings?.workspaceClass || "g1-standard",
+    );
+
+    const handleSubmit = useCallback(
+        (e: FormEvent) => {
+            e.preventDefault();
+
+            const newInterval = Math.abs(Math.min(Number.parseInt(interval), 100)) || 0;
+
+            const updatedConfig: PartialConfiguration = {
+                configurationId: configuration.id,
+                prebuildSettings: {
+                    ...configuration.prebuildSettings,
+                    prebuildInterval: newInterval,
+                    branchStrategy,
+                    branchMatchingPattern,
+                    workspaceClass,
+                },
+            };
+
+            updateConfiguration.mutate(updatedConfig, {
+                onSuccess: () => {
+                    toast("Your prebuild settings were updated.");
+                },
+            });
+        },
+        [
+            branchMatchingPattern,
+            branchStrategy,
+            configuration.id,
+            configuration.prebuildSettings,
+            interval,
+            toast,
+            updateConfiguration,
+            workspaceClass,
+        ],
+    );
+
+    // TODO: Figure out if there's a better way to deal with grpc enums in the UI
+    const handleBranchStrategyChange = useCallback((val) => {
+        // This is pretty hacky, trying to coerce value into a number and then cast it to the enum type
+        // Would be better if we treated these as strings instead of special enums
+        setBranchStrategy(parseInt(val, 10) as BranchMatchingStrategy);
+    }, []);
+
+    return (
+        <ConfigurationSettingsField>
+            <form onSubmit={handleSubmit}>
+                <Heading3>Prebuild Settings</Heading3>
+                <Subheading className="max-w-lg">These settings will be applied on every Prebuild.</Subheading>
```

Comment:
> These settings will be applied on every Prebuild.

This is more of an aside since it's relevant to all of the work we've been doing with Configs so far: do we or do we not capitalize feature names? I like the aesthetics of doing so, but am not convinced too hard. Either way, it would be great to settle this so that we can make it consistent. 

cc @loujaybee since you may have an opinion :)